extensions/wikia/EmbeddableDiscussions/EmbeddableDiscussionsApiController.class.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+class EmbeddableDiscussionsApiController extends WikiaApiController {
+	const SORT_BY_WIKIFACTORY = 'wgEmbeddableDiscussionsSortBy';
+	const SELECTED_CATEGORIES_WIKIFACTORY = 'wgEmbeddableDiscussionsSelectedCategories';
+
+	const SORT_BY_KEY = "sortBy";
+	const ALL_CATEGORIES_KEY = 'allCategories';
+	const SELECTED_CATEGORIES_KEY = 'selectedCategories';
+
+	const SORT_BY_LATEST = "latest";
+	const SORT_BY_TRENDING = "trending";
+
+	private $categories;
+
+	public function __construct() {
+		global $wgCityId;
+
+		parent::__construct();
+
+		$this->categories = ( new DiscussionsThreadModel( $wgCityId ) )->getAllCategories();
+	}
+
+	public function readSortBy() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)